### PR TITLE
CJ-8 - Use new Google authentication service

### PR DESCRIPTION
**Pull request by andresmoschini**
Hi @AlphaGit, @jwaimann, @mbeckerle or _another generous reviewer_,

I have migrated authentication services to Google+ OAuth removing all dependencies, using only raw requests because the [previous approach](https://github.com/CommonJobs/CommonJobs/pull/373) fell into a dependency hell. 

Could you review my changes?

### CODE REVIEW

**Comment by andresmoschini** [comment]
@AlphaGit

**Comment by AlphaGit** [comment]
Hi! I'll review today, thanks for looking into that.

File: source/CommonJobs/CommonJobs.MVC.UI/Infrastructure/OAuthHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            var uriBuilder = new UriBuilder(url)
+            {
+                Path = path,
+                Query = ConstructQueryString(query)
+            };
+            return uriBuilder.ToString();
+        }
+
+        public static String ConstructQueryString(NameValueCollection parameters, string separator = "&")
```
Comment (AlphaGit) [inline source/CommonJobs/CommonJobs.MVC.UI/Infrastructure/OAuthHelpers.cs]: (Not important) Does it make sense to have the separator as a parameter? Will it ever be different than `&`?

File: source/CommonJobs/CommonJobs.Mvc.UI/Web.config [hunk 2]
```diff
@@ -17,6 +17,8 @@
     <add key="CommonJobs/DocumentationBaseUrl" value="http://commonjobs.makingsense.com/documentation" />
     <add key="LOGENTRIES_ACCOUNT_KEY" value="" />
     <add key="LOGENTRIES_LOCATION" value="" />
+    <add key="CommonJobs/GoogleOAuthClientId" value="293136734985.apps.googleusercontent.com" />
+    <add key="CommonJobs/GoogleOAuthSecret" value="z0IsQwpgFZBTRsmvSvzdThKV" />
```
Comment (AlphaGit) [inline source/CommonJobs/CommonJobs.Mvc.UI/Web.config]: Should this be in the public repository?

**Comment by AlphaGit** [comment]
Aside from my comments above, the changes look great. I slightly disagree with the tendency of assigning everything to a variable, even when it's used only for being returned in a method -- but I don't think it's something important.

Great job! And thanks for cleaning all the dependencies!

File: source/CommonJobs/CommonJobs.MVC.UI/Infrastructure/OAuthHelpers.cs (hunk 1 above)
Comment (andresmoschini) [inline source/CommonJobs/CommonJobs.MVC.UI/Infrastructure/OAuthHelpers.cs]: You are right, but this code is copied from another codebase, I think that it is better to keep it as it is.

File: source/CommonJobs/CommonJobs.Mvc.UI/Web.config (hunk 2 above)
Comment (andresmoschini) [inline source/CommonJobs/CommonJobs.Mvc.UI/Web.config]: No, but it is not so important, it is my personal Google API secret, if someone stole it, the worst that could happen is that he consume all the quota, but it is really difficult. 
Of course if that happen, I will generate a new secret and change it outside code repository.

**Comment by andresmoschini** [comment]
Thanks @AlphaGit and @matiasbeckerle, I will publish it tomorrow and CommonJobs will continue working after April 20

File: source/CommonJobs/CommonJobs.MVC.UI/Infrastructure/OAuthHelpers.cs (hunk 1 above)
Comment (AlphaGit) [inline source/CommonJobs/CommonJobs.MVC.UI/Infrastructure/OAuthHelpers.cs]: Agreed!

**Pull request merged**